src/anonymous-credentials/schema.ts

```diff
@@ -1101,13 +1102,19 @@ export class CredentialSchema extends Versioned {
         } else if (schemaProps[key]['type'] == 'object' && typ == 'object') {
           const schemaKeys = new Set([...Object.keys(schemaProps[key]['properties'])]);
           const valKeys = new Set([...Object.keys(value)]);
-          for (const vk of valKeys) {
-            CredentialSchema.generateFromCredential(value, schemaProps[key]['properties']);
-          }
-          // Delete extra keys not in cred
-          for (const sk of schemaKeys) {
-            if (value[sk] === undefined) {
-              delete schemaKeys[sk];
+
+          // If empty object, skip it here otherwise causes problems downstream
+          if (schemaKeys.size === 0) {
+            delete schemaProps[key];
```

Comment:
> This isn't needed as it will ignore credential values for which the schema has an empty object type

that is intended, because that value cant exactly be encoded (unless its stringified to {} but that seems pointless to me?)

Reply:
Why? The schema should never have an empty object.